### PR TITLE
simplify logic of containerAndPodFromLabels func

### PR DESCRIPTION
**Pull request by 249043822**
Removing the else primitive to make the code simple

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/32760)

### CODE REVIEW

File: pkg/kubelet/dockertools/docker_manager.go [hunk 1]
```diff
@@ -1586,19 +1586,17 @@ func containerAndPodFromLabels(inspect *dockertypes.ContainerJSON) (pod *api.Pod
 			if container == nil {
 				err = fmt.Errorf("unable to find container %s in pod %v", name, pod)
 			}
-		} else {
-			pod = nil
+			// As pod could't be nil, need not run the code below, we return directly here
```
Comment (MHBauer) [inline pkg/kubelet/dockertools/docker_manager.go]: because it's assigned a default struct on L#1569?

**Review by MHBauer — COMMENTED** [review]
Seems to be a valid transformation of the existing code. I'm not sure the existing code made any sense.

File: pkg/kubelet/dockertools/docker_manager.go (hunk 1 above)
Comment (MHBauer) [inline pkg/kubelet/dockertools/docker_manager.go]: and because if we're inside the `if`, it's decoded successfully.

Comment (yujuhong) [inline pkg/kubelet/dockertools/docker_manager.go]: I'd rather not change this. Returning in a nested, long `if` block hinders the readability.

**Comment by 249043822** [comment]
Thanks for reviwing @yujuhong @MHBauer , I will close it.

**Pull request closed without merging**